src/main/rx/spektrum.c

```diff
@@ -139,10 +142,21 @@ static uint8_t spektrumFrameStatus(rxRuntimeConfig_t *rxRuntimeConfig)
         }
     }
 
-    /* only process if srxl enabled, some data in buffer AND servos in phase 0 */
-    if (srxlEnabled && telemetryBufLen && (spekFrame[2] & 0x80) == 0) {
-        dispatchAdd(&srxlTelemetryDispatch, SPEKTRUM_TELEMETRY_FRAME_DELAY);
+#if defined(USE_TELEMETRY) && defined(USE_TELEMETRY_SRXL)
+    if (srxlEnabled) {
+        /* Only dispatch for transmission if there are some data in buffer AND servos in phase 0 */
+        if (telemetryBufLen && (spekFrame[2] & 0x80) == 0) {
+            dispatchAdd(&srxlTelemetryDispatch, SPEKTRUM_TELEMETRY_FRAME_DELAY);
+        }
+
+        /* Trigger tm data collection if buffer has been sent and is empty, 
+           so data will be ready to transmit in the next phase 0 */
+        if (telemetryBufLen == 0) {
+            srxlCollectTelemetryNow();
```

Comment:
Would this (or rather the whole block) be a candidate to be moved into `processFrameFn` (see https://github.com/betaflight/betaflight/blob/7ad4bbfa0fbc4920938997a75cf4da7cfab4c66d/src/main/rx/fport.c#L328-L351). This way it will execute with a more appropriate priority.

Reply:
Well, I can not directly see how SpektrumRC/SRXL halfduplex RC/TM could be combined with fport/SmartPort.  Seems like two different worlds, from what I can tell. But then I have not studied neither f- or S-port to say anything really. As of now, and same as earlier, SRXL, is a two stage operation. Collecting the data and sending it. Both happening in other tasks from RX handling, data collection in the TELEMETRY task and sending the block it in a DISPATCHed task. That is how I understand it, I have not done that nor changed any of it. The main change is from a free running asyncronous 30Hz task to a task synchronized with and triggered by the RX task. The actual TM transmission is still the same as before, dispatched after a delay after receiving phase 0 RC frame. (As per Spektrum specs). If equal amounts of phase 0 and 1 frames are received, the effective TM rate will be half the RC rate. Or 22mS /45Hz with a 11mS/90Hz frame rate. that is 50% faster than the current 30Hz rate. But then there are also some jitter delays to add, as the current 30Hz are asynchronously free running. Now we are synced and locked. And in that situation, we could combine data collection and transmission into one and the same tasks. But I find it rather nice to have things separated liked it is. I am not sure how that could be combined with f/s-port either.
But that could perhaps be a next step after merging this PR......

Reply:
Ah, I didn't mean to combine anything with FPort / SBus, I just referenced it as an example of how task separation inside a `rx.c` driver can be done.

Instead of running telemetry collection and sending outside of `spektrum.c` and using custom triggers to trigger it, you could put the code for these tasks into an implementation of `processFrameFn`. Then `spektrumFrameStatus` will only have to set `RX_FRAME_PROCESSING_REQUIRED` on the return value, and your `processFrameFn` will be called with the right priority by the scheduler.

Reply:
OK, sorry. Misunderstood the comment. Might be possible to align the way TM data collection and transmission is done. I have to study a bit more to get the grips on how it is done in F/S-port. And whether it is better or not. But of course, doing things in the same or similar way is always a good thing. 
Could we please merge this PR as is for  now, and do a possible alignment later on?

Reply:
Yeah, no problem, this can always be done later.